File: 10-19/19/main.py
def is_leap_year(year):
    return (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0)

def count_sundays():
    days_in_months = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    day_of_week = 1 
    sundays_on_first = 0

    for year in range(1900, 1901):
        for month in range(12):
            day_of_week = (day_of_week + days_in_months[month] + (1 if month == 1 and is_leap_year(year) else 0)) % 7

    for year in range(1901, 2001):
        for month in range(12):
            if day_of_week == 0:
                sundays_on_first += 1
            day_of_week = (day_of_week + days_in_months[month] + (1 if month == 1 and is_leap_year(year) else 0)) % 7

    return sundays_on_first

print("Number of Sundays:", count_sundays())

File: 10-19/19/main.cpp
#include <iostream>
using namespace std;

bool isLeapYear(int year) {
    return (year % 4 == 0 && year % 100 != 0) || (year % 400 == 0);
}

int main() {
    int daysInMonths[] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    int dayOfWeek = 1;
    int sundaysOnFirst = 0;

    for (int year = 1900; year < 1901; year++)
        for (int month = 0; month < 12; month++)
            dayOfWeek = (dayOfWeek + daysInMonths[month] + (month == 1 && isLeapYear(year) ? 1 : 0)) % 7;

    for (int year = 1901; year <= 2000; year++) {
        for (int month = 0; month < 12; month++) {
            if (dayOfWeek == 0) sundaysOnFirst++;
            dayOfWeek = (dayOfWeek + daysInMonths[month] + (month == 1 && isLeapYear(year) ? 1 : 0)) % 7;
        }
    }

    cout << sundaysOnFirst << endl;
    return 0;
}